modules/page_base.py

```diff
@@ -570,6 +583,19 @@ def switch_to_new_tab(self) -> Page:
         self.driver.switch_to.window(handles[-1])
         return self
 
+    def switch_to_alert(self) -> Page:
+        """Wait for an alert to appear, then switch to it"""
+
+        def alert_present(driver) -> bool:
+            try:
+                driver.switch_to.alert
+                return True
+            except NoAlertPresentException:
```

Comment:
Is it up to the user using this method to error check and raise the exception (if the alert isnt present)? I was wondering because I don't see any instances of it in the code

Reply:
Yeah, I ended up not needing it. I guess it really doesn't strictly fail if there's no alert, which is less than desirable. I think I might just delete this function